### DISCUSSION TITLE
Get site logo and display in WPF app.

### DISCUSSION THREAD

**Discussion by KarolFilipKozlowski** [pnp/pnpcore#1152] [General]
Hello, 
I try to build a small wpf app where I have some information about a given site. One of the features that I want is to show website log.
In debug, I have logo url, but isn't in the app, I think the app cannot process bitmap/image/logo because haven't directly access because downloading and displaying the png works outside context, so its any way to get that file?

````
{
    var myWeb = await context.Web.GetAsync(p => p.Title, p => p.Description, p => p.SiteLogoUrl, p => p.SiteLogoDescription);

    BitmapImage bitmap = new();
    bitmap.BeginInit();
    bitmap.UriSource = new Uri($@"https://{uri.Host}{myWeb.SiteLogoUrl}", UriKind.RelativeOrAbsolute);
    bitmap.EndInit();

    Image image = new()
    {
        Source = bitmap,
        ToolTip = myWeb.SiteLogoDescription
    };

    img_SiteLogo.Source = bitmap;
    lbl_SiteTitle.Content = myWeb.Title;
    lbl_SiteDescription.Content = myWeb.Description;
}
````
![image](https://user-images.githubusercontent.com/25957036/229286220-35627d1e-3ba2-4243-a270-22828d8ee63f.png)

**Comment by jansenbe** [comment]
@KarolFilipKozlowski : I think you've to download the logo first before you can use/show it. When the WPF application is showing the bitmap image it will not know how to authenticate against SharePoint Online. Did not test any of this, just think this might be reason

**Reply by KarolFilipKozlowski** [reply]
Yes, I tried to download/save, and I got only an empty file (0 bytes), but I was a just test, and I'm not good with that operation in C#, so maybe I did it in the wrong way. Anyway, I will be tried in future.
